src/test/scala/com/modelfabric/test/FusekiManager.scala

```diff
@@ -89,6 +91,9 @@ class FusekiManager(val endpoint: HttpEndpoint) extends Actor with ActorLogging
         fusekiRunner.startServer()
       }
       /* start pinging the server and issue StartOk to the sender when ping is successful */
+      // JC: it's not clear that the logic is to ping fureki 10 times to check if it's available.
+      // I think it's better to define two states for the actor. The Ping class is used for multiple purpose,
+      // with multiple parameters to control behavior, not straightforward to understand
```

Comment:
True, but again, this is a test class that does the work for us already. I would not worry about making it more understandable, unless you really think it would add more value?